pkg/kubelet/preemption/preemption.go

```diff
@@ -111,6 +112,11 @@ func (c *CriticalPodAdmissionHandler) evictPodsToFreeRequests(admitPod *v1.Pod,
 			// In future syncPod loops, the kubelet will retry the pod deletion steps that it was stuck on.
 			continue
 		}
+		if len(insufficientResources) > 0 {
+			metrics.Preemptions.WithLabelValues(insufficientResources[0].resourceName.String()).Inc()
```

Comment:
Can you share more about why only the first entry in the list is being used as a label?

Reply:
Does it relate to your comment: `A preemption is only recorded for one resource`? Can you share more about that decision?

Reply:
Mostly due to cardinality - otherwise we would have one potential label per type, and on contended nodes that could be a large number.  If we don’t allow that unbounded cardinality, then we can’t easily get an accurate count by summing across reasons.  So trying to include the signal without potentially going overboard.  I think if we wanted more granular preemption metrics we might design this slightly differently, but it might be best to do that from kube-state-metrics or the scheduler, which can also record more of the reasons why.